chiselFrontend/src/main/scala/chisel3/core/Assert.scala

```diff
@@ -51,12 +51,15 @@ object assert { // scalastyle:ignore object.name
   }
 
   def apply_impl_do(cond: Bool, line: String, message: Option[String], data: Bits*)(implicit sourceInfo: SourceInfo) {
+    val escLine = line.replaceAll("%", "%%")
     when (!(cond || Builder.forcedReset)) {
       val fmt = message match {
-        case Some(str) => s"Assertion failed: $str\n    at $line\n"
-        case None => s"Assertion failed\n    at $line\n"
+        case Some(str) =>
+          val msg = if (data.isEmpty) str.replaceAll("%", "%%") else str
```

Comment:
The functions will still have the same name so the effect is the same. Having another function signature seems superfluous as you are just having the compiler do the dispatching for you.

In either case this will continue to work:
assert(???: Bool, "1 % 0 is undefined.")